File: c/test/test_basic.c
#include "test_utils.h"
#include <emtrace/emtrace.h>

EXPECT_OUTPUT(
    "An integer: 42\n"
    "A string: a string\n"
);

int main(void) {
    EMTRACE_INIT();
    EMTRACELN("Hello from the basic test!");
    int x = 42;
    EMTRACELN_F("An integer: {}", int, x);
    const char* s = "a string";
    EMTRACE("A string: ");
    EMTRACELN_S(s);
    return 0;
}

File: rust/emtrace/examples/test_basic.rs
use emtrace::{emtrace_init, expect, trace, traceln};

fn main() {
    expect!(
        "Hello from the basic test!\n\
         An integer: 42\n\
         A string: a string\n"
    );
    emtrace_init!().unwrap();

    traceln!("Hello from the basic test!");
    let x: i32 = 42;
    traceln!("An integer: {}", i32: x);
    let s: &str = "a string";
    trace!("A string: ");
    traceln!("{}", str: s);
    // traceln!("A list: {}", [i32]: vec![1, 2, 3, 4]);
}
